crates/contracts/solidity/PhonyERC20.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
```

Comment:
This facade "phony" token enables our simulation to mint tokens to arbitrary users. Thus, we no longer need to find a token holder with sufficient balance, and can simulate a settlement by minting tokens. It proxies all other calls to the actual token implementation (and will try to transfer with the actual token balance before using internal balances).

The motivation for using this technique over the existing "token holder finder", is that it is possible that the token holders we find end up getting used as part of the settlement (for example, if the token holder we find is a Uniswap V2 pool, and the trade goes over the pool on Uniswap V2 - creating a token imbalance and causing a potentially legitimate trade path to not work).

The downside is that we don't always exercise the token's `transfer` function - but there is a proposed solution to this in the PR description.